openassessment/staffgrader/staff_grader_mixin.py

```diff
@@ -194,6 +201,39 @@ def list_staff_workflows(self, data, suffix=''):  # pylint: disable=unused-argum
                 log.exception("Failed to serialize workflow %d: %s", staff_workflow.id, str(e), exc_info=True)
         return result
 
+    @XBlock.json_handler
+    @require_course_staff("STUDENT_GRADE")
+    def list_assessments(self, data, suffix=''):  # pylint: disable=unused-argument
+        """
+        List the assessments grades based on the type (received or given) for a specific submission.
+
+        Args:
+            data (dict): Contains the necessary information to fetch the assessments.
+                - 'item_id': The ID of the xblock/item.
+                - 'submission_uuid': The UUID of the submission.
+                - 'assessment_type': A string, either "received" or any other
+                    value to determine the type of assessments to retrieve.
```

Comment:
Hi @BryanttV , thank you for submitting these PRs! I've only started reviewing, and should have a full review ready for you tomorrow.

From the PR description:

> ### Considerations
> In the new /api/ora_staff_grader/assessments/feedback endpoint there is a query parameter that is used to filter the assessment grades based on the type (received or given) for a specific submission.
> 
> For the name of this query parameter we had 2 possible proposals:
> 
>   * AssessmentType: We think this is a clear name, however, from ORA there is already a reference to this name to classify step types such as Self, Staff, Peer, and Training.
>   * AssessmentFilter: There is no reference to this name from ORA, but it may be too generic and does not clearly describe what the function of the query parameter is.
> 
> Does anyone have suggestions for a clearer, more descriptive name for this query parameter?

Ya, that's tricky.. but since your [handler is actually just an "if/else" statement wrapping these two options](https://github.com/openedx/edx-ora2/pull/2101/files#diff-325ab5d2eef948ecf0b39bbcda9b0b2071dd1c38f9a3eafa3d0e178413a52dedR230-R235), would you consider using two separate endpoints, e.g.:

* /api/ora_staff_grader/assessments/feedback/from?submissionUUID=...&oraLocation=...
* /api/ora_staff_grader/assessments/feedback/to?submissionUUID=...&oraLocation=...

Reply:
> Including scorer information and scoring details to fill the following Staff Grade MFE table:

Is there an MFE somewhere that shows this API in use?

Reply:
Nudge @BryanttV ^

CC @felipemontoya @mariajgrimaldi

Reply:
Oh sorry! I'm just seeing this comment. I like the from/to idea since we're not using complex query parameters. And yes! We're using this PR to feed a table in the ORA MFE: https://github.com/openedx/frontend-app-ora-grading/pull/263

@BryanttV: could you link the ORA MFE PR in the cover letter? Also, what do you think about the suggestion? Does it work alongside the ORA Grading MFE?

@pomegranited: we'll find some time this week to address your comments in detail. Thank you for the review!

Reply:
@mariajgrimaldi Thank you for your reply! I wanted to make sure this change was still needed before doing any further review, and hadn't heard back from @BryanttV .

Let me know when this is ready for another look?

Reply:
Hi @pomegranited, Thanks for your comments and suggestions! and sorry for not responding sooner. I'm currently making the necessary changes to make the PRs ready for review.

Reply:
No worries at all @BryanttV, just drop me a message here when you're ready.

And thank you for the link to that MFE, that will help with testing.

Reply:
Hi again @pomegranited, I have already made the refactor to separate the endpoints. I updated the cover letter so you can test the endpoints directly following the instructions.

For the moment, you cannot test using the MFE as it does not yet have the corresponding modifications using the current refactoring. I will be attentive to any questions or comments, thank you very much!